libs/sdk-ui-dashboard/src/model/queries/kpis.ts

```diff
@@ -0,0 +1,68 @@
+// (C) 2021 GoodData Corporation
+import { IDashboardQuery } from "./base";
+import { ObjRef } from "@gooddata/sdk-model";
+import { ICatalogDateDataset } from "@gooddata/sdk-backend-spi";
+
+/**
+ * Given a reference to a measure, this query will obtain list of all date datasets that may be used
+ * to filter it.
+ *
+ * @alpha
+ */
+export interface QueryMeasureDateDatasets extends IDashboardQuery<MeasureDateDatasets> {
+    readonly type: "GDC.DASH/QUERY.MEASURE.DATE.DATASETS";
+    readonly payload: {
+        readonly measureRef: ObjRef;
+    };
+}
+
+/**
+ * Creates action through which you can query dashboard component for information about the date data sets
+ * that are applicable for filtering of the provided measure. measures are typically used to obtain value to render on KPIs.
+ *
+ * @param measureRef - reference to measure
+ * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
+ *  events that will be emitted during the command processing
+ * @alpha
+ */
+export function queryDateDatasetsForMeasure(
+    measureRef: ObjRef,
+    correlationId?: string,
+): QueryMeasureDateDatasets {
+    return {
+        type: "GDC.DASH/QUERY.MEASURE.DATE.DATASETS",
+        correlationId,
+        payload: {
+            measureRef,
+        },
+    };
+}
+
+/**
+ * The measure date datasets holds information about available date datasets that can be used for date-filtering a particular
+ * measure.
+ *
+ * The data included herein can be used to select an appropriate date dataset to filter a KPI widget that renders value
+ * of particular measure.
+ *
+ * @alpha
+ */
+export type MeasureDateDatasets = {
+    /**
+     * Date datasets that are available for filtering of the measure. The available datasets are obtained by inspecting
+     * relation of measure and the different date datasets in the workspace's logical data model.
+     */
+    readonly dateDatasets: ReadonlyArray<ICatalogDateDataset>;
```

Comment:
Do we need both `dateDatasets` and `dateDatasetsOrdered`? Could we use just `dateDatasets` and make sure they are ordered?

Reply:
no technical problem doing that. although my thinking was:

- dateDatasets contains the raw data as returned by backend
- dateDatasetsOrdered contains datasets ordered using the dashboards 'arbitrary' ordering logic

the presentation layer should use the arbitrarily ordered data sets however in case someone wants the data sets as returned by backend then they are there.

e.g. full data 'disclosure' & hiding of the arbitrary sorting logic (which was previously in goodstrap/kit)

Reply:
Ok, that is fair.